source/backend/tests/unit/api/Services/ProjectServiceTest.cs

```diff
@@ -66,6 +65,83 @@ public void Search_NoPermission()
             act.Should().Throw<NotAuthorizedException>();
             repository.Verify(x => x.SearchProjects(It.IsAny<string>(), It.IsAny<int>()), Times.Never);
         }
-        #endregion
+
+        [Fact]
+        public void Search_GetPage_ShouldFail_NotAuthorized()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+            var service = helper.Create<ProjectService>(user);
+
+            var repository = helper.GetService<Mock<IProjectRepository>>();
+
+            // Act
+            Action act = () => service.GetPage(new ProjectFilter {  ProjectName = "test" });
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+            repository.Verify(x => x.GetPageAsync(It.IsAny<ProjectFilter>()), Times.Never);
+        }
+
+        [Fact]
+        public void Search_GetPage_ShouldFail_Filter_IsNull()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView);
+            var service = helper.Create<ProjectService>(user);
+
+            var repository = helper.GetService<Mock<IProjectRepository>>();
+
+            // Act
+            Action act = () => service.GetPage(null);
+
+            // Assert
+            act.Should().Throw<ArgumentException>();
+            repository.Verify(x => x.GetPageAsync(It.IsAny<ProjectFilter>()), Times.Never);
+        }
+
+        [Fact]
+        public void Search_GetPage_ShouldFail_Filter_IsInvalid()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView);
+            var service = helper.Create<ProjectService>(user);
+
+            var repository = helper.GetService<Mock<IProjectRepository>>();
+
+            // Act
+            Action act = () => service.GetPage(new ProjectFilter { Page = 0 });
+
+            // Assert
+            act.Should().Throw<ArgumentException>();
+            repository.Verify(x => x.GetPageAsync(It.IsAny<ProjectFilter>()), Times.Never);
+        }
+
+        [Fact]
+        public async void Search_GetPage_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.ProjectView);
+            var service = helper.Create<ProjectService>(user);
+
+            var repository = helper.GetService<Mock<IProjectRepository>>();
+            repository.Setup(x => x.GetPageAsync(It.IsAny<ProjectFilter>()))
+                .ReturnsAsync(new Paged<PimsProject>()
+                {
+                    Page = 1,
```

Comment:
is the page necessary?

Reply:
Yes, it would throw an excpetion otherwise.